prowler/providers/azure/azure_provider.py

```diff
@@ -244,14 +380,86 @@ def setup_session(
 
         return credentials
 
-    def __check_service_principal_creds_env_vars__(self):
+    @staticmethod
+    def test_connection(
+        az_cli_auth,
+        sp_env_auth,
+        browser_auth,
+        managed_identity_auth,
+        tenant_id,
+        region,
+        credentials=None,
+    ) -> tuple[DefaultAzureCredential, AzureRegionConfig]:
+        """
+        Test the connection to an Azure subscription using the provided credentials.
+
+        Args:
+            credentials: The credentials object used to authenticate with Azure.
+            az_cli_auth (bool): Flag indicating whether to use Azure CLI authentication.
+            sp_env_auth (bool): Flag indicating whether to use Service Principal authentication with environment variables.
+            browser_auth (bool): Flag indicating whether to use interactive browser authentication.
+            managed_identity_auth (bool): Flag indicating whether to use managed identity authentication.
+            tenant_id (str): The Azure Active Directory tenant ID.
+            region (str): The Azure region.
+
+        Returns:
+            tuple: A tuple containing the credentials and region configuration objects.
+        """
+        try:
+            region_config = AzureProvider.setup_region_config(
+                validate_azure_region(region)
+            )
+
+            # Set up the Azure session
+            if not credentials:
+                credentials = AzureProvider.setup_session(
+                    az_cli_auth,
+                    sp_env_auth,
+                    browser_auth,
+                    managed_identity_auth,
+                    tenant_id,
+                    region_config,
+                )
+            # Create a SubscriptionClient
+            subscription_client = SubscriptionClient(credentials)
+
+            # Get info from the first subscription
+            subscription = next(subscription_client.subscriptions.list())
+
+            logger.info(f"Connected to Azure subscription: {subscription.display_name}")
+            return credentials, region_config
+
+        except HttpResponseError as error:
+            logger.error(
+                f"{error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+            )
+            raise error
```

Comment:
Can you rename this error to `credentials_error`?

Reply:
Done